### ISSUE TITLE
TextView slow when highlighting in text with a lot of styling

### ISSUE THREAD

**Issue by normen** [rivo/tview#533]
Hi,

in my app I am using jp2a to display images in a text window. This happens through the ANSIWriter adapter and works fine. jp2a generates colored "ASCII Art" from the images. However when the TextView displays about 5 or 10 of such images (not too many characters really) then setting highlights in that TextView become slower and slower. I noticed that its probably the highlights because I can scroll the textView just fine, only setting a highlight is slow.

Cheers for this library, its awesome!
Normen

**Comment by rivo** [comment]
Yeah, I'm guessing the ANSIWriter creates a lot of extra tags which then have to be parsed again and translated back into colours, line breaking and all. I would expect this to get slow the more you add.

I'm not sure I have a quick fix for this. But I do have an `Image` primitive in mind for the future that will do something similar to  `jp2a`. That way, you won't have to go through the `TextView`. Implementing it would mean ignoring issues for a while, though, so I'm not yet sure where to set priorities.

**Comment by normen** [comment]
I see, thanks for the answer. I saw that the matrix client [gomuks](https://github.com/tulir/gomuks) has a nice ANSI-Image conversion thingy, maybe thats interesting.

**Comment by rivo** [comment]
Closing this as I don't see any solution to this issue. Performance will get better with improvements made to `TextView` but the double-conversion will always remain and potentially slow the application down.
